### PR TITLE
Various additions & clean-ups for UnitTests and README file TODOs

### PR DESCRIPTION
**Pull request by andreicozma1**
Added checks for token file in setup class for unit tests.

Added TODO's in the README.md file

### CODE REVIEW

**Review by tylerbeichler — APPROVED** [review]

File: tests/main_test.py [hunk 1]
```diff
@@ -151,23 +156,23 @@ def test_group_no_desc(self):
...
-        assert content['data']['name'] == 'test group name'
+        response = self.do_post('/group/info', {'id': self.group['_id']['$oid']})
+        self.ensure_status_code_msg(response, 200, "Group successfully retrieved.")
+        assert response.json()['data']['name'] == 'test group name'
 
         # delete the group
-        content, status_code = self.do_post('/group/delete', {'id': self.group['_id']['$oid']})
-        assert status_code == 200
+        response = self.do_post('/group/delete', {'id': self.group['_id']['$oid']})
+        self.ensure_status_code_msg(response, 200, "Group successfully deleted.")
```
Comment (andreicozma1) [inline tests/main_test.py:170]: @tdbucks after this you may also want to add another check that calls /group/info

File: tests/main_test.py [hunk 2]
```diff
@@ -33,61 +50,50 @@ def teardown_class(cls):
...
         test
         :return:
         """
-
         # delete
-        content, status_code = self.do_post('/user/delete', {'sub': self.user['data']['sub']})
-        assert status_code == 200
-        assert content['msg'] == "User profile successfully deleted."
+        response = self.do_post('/user/delete', {'sub': self.user['data']['sub']})
+        self.ensure_status_code_msg(response, 200, "Successfully deleted the user profile.")
```
Comment (andreicozma1) [inline tests/main_test.py:75]: @tdbucks after this check you may also want to have test to /user/info reporting that the user doesn't exist

File: tests/main_test.py [hunk 3]
```diff
@@ -33,61 +50,50 @@ def teardown_class(cls):
...
+        response = self.do_post('/user/delete', {'sub': self.user['data']['sub']})
+        self.ensure_status_code_msg(response, 404, "Token is unauthorized or user does not exist.")
+        
         # register new user for status_code 201
-        content, status_code = self.do_post('/register', {})
-        assert status_code == 201
-        assert content['data']['sub'] == self.user['data']['sub']
+        response = self.do_post('/register', {})
+        self.ensure_status_code_msg(response, 201, "User successfully retrieved.")
+        assert response.json()['data']['sub'] == self.user['data']['sub']
```
Comment (andreicozma1) [inline tests/main_test.py:84]: Probably safest to add another /user/info here again

File: tests/main_test.py [hunk 4]
```diff
@@ -97,8 +103,8 @@ def test_bad_payment(self):
                 'preferred': 'cashapp'
             }
```
Comment (andreicozma1) [inline tests/main_test.py:104]: add test cases for overly long values for each (make sure the API has some kind of max length as well)
Also, perhaps we shouldn't allow spaces in the username either.

File: tests/main_test.py [hunk 5]
```diff
@@ -97,8 +103,8 @@ def test_bad_payment(self):
                 'preferred': 'cashapp'
```
Comment (andreicozma1) [inline tests/main_test.py:103]: Add another test case for only updating "preferred" where the value supplied isn't actually allowed.
For example "preferred": "non-existent" should not be successful

File: tests/main_test.py [hunk 6]
```diff
@@ -194,46 +198,47 @@ def test_create_group(self):
...
 
         # check persons groups
-        content, status_code = self.do_post('/user/info', {'sub': self.user['data']['sub']})
-        assert status_code == 200
-        assert self.group['_id'] not in content['data']['groups']
+        response = self.do_post('/user/info', {'sub': self.user['data']['sub']})
+        self.ensure_status_code_msg(response, 200, "User successfully retrieved.")
+        assert self.group['_id'] not in response.json()['data']['groups']
 
     def test_invite(self):
```
Comment (andreicozma1) [inline tests/main_test.py:243]: There's two different tests made here (invite while creating the group, and also invite after creating the group with /group/invite)
Separate them into two different test cases (two different functions)

**Pull request merged**